--- a/server/scheduler.c.orig	2020-06-02 00:40:52.767051080 -0400
+++ b/server/scheduler.c	2020-06-02 00:40:37.257957277 -0400
@@ -114,7 +114,7 @@
     if (thread->priority >= THREAD_PRIORITY_TIME_CRITICAL)
     {
         policy = SCHED_FIFO;
-        param.sched_priority = thread_base_priority + 4;
+        param.sched_priority = thread_base_priority + 10;
     }
     else if (thread->priority >= THREAD_PRIORITY_HIGHEST)
     {

 
